### PR TITLE
Add github action to run Flake8 on PRs and commits to main

### PR DESCRIPTION
**Pull request by Jkrzy**
Adds action to run Python linting with Flake8 on PRs and commits to main.

### CODE REVIEW

**Review by NicholasKramer — APPROVED** [review]
Approved, nice linter addition!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -10,6 +10,27 @@ on:
...
+      - name: Install Flake8
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+      - name: Lint with flake8
+        working-directory: ./backend
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
```
Comment (tadhg-ohiggins) [inline .github/workflows/test.yml:31]: Is this line referring to something I'm missing?

Comment (Jkrzy) [reply .github/workflows/test.yml:31]: Good catch, leftover comment from dev. I can simplify this a bit!
```suggestion
          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -10,6 +10,27 @@ on:
...
+          flake8 . --count --max-complexity=10  --statistics
```
Comment (Jkrzy) [inline .github/workflows/test.yml:32]: Combining flake8 calls and looking not selecting specific errors to search for here.
```suggestion
          flake8 . --count --max-complexity=10  --show-source --statistics
```

**Comment by Jkrzy** [comment]
@tadhg-ohiggins I made a few small tweaks here to clean up a few linter reports, merging on existing approvals

**Pull request merged**